### PR TITLE
Tenant choice (interactive login authentication)

### PR DESCRIPTION
**Pull request by cdrcc**
It is not possible to chose a tenant during authentication process, which is an issue when you want to access another one with the same login : the default one remains selected. This feature is adding the opportunity to define a "tenant_id" argument while loging in with the interactive login authentication.

### CODE REVIEW

**Comment by cdrcc** [comment]
@microsoft-github-policy-service agree

File: powerbiclient/authentication.py [hunk 1]
```diff
@@ -104,27 +104,34 @@ def _acquire_token_device_code(self):
...
         """Returns the authentication result captured using interactive login
 
         Returns:
             dict: authentication result
         """
-        app = msal.PublicClientApplication(client_id=CLIENT_ID)
+        if not tenant_id:
+            app = msal.PublicClientApplication(client_id=CLIENT_ID)
+        else:
+            app = msal.PublicClientApplication(client_id=CLIENT_ID, authority=f"https://login.microsoftonline.com/{tenant_id}")
```
Comment (orshemesh16) [inline powerbiclient/authentication.py:129]: Please apply the changes also for device code login flow. In addition, add the authority template string in the top of the file to be reused for both flows.

**Review by orshemesh16 — CHANGES_REQUESTED** [review]
The change was approved but need to apply it also for the second authentication flow (see my comment). Please fix it so we could complete the PR.

**Review by orshemesh16 — APPROVED** [review]

**Comment by orshemesh16** [comment]
Hi @cdrcc, thank you for your contribution. 
This change will be included in the next patch version.

**Pull request merged**